ConsoleGame.java

```diff
@@ -77,7 +73,18 @@ public void gameIsOver(Board board, Player winner) {
 
     @Override
     public String userPickGameMode() {
-        Scanner s = new Scanner(System.in);
-        return s.nextLine();
+        String input = null;
+        try {
+            input = reader.readLine();
+            for (Gamemodes gm : Gamemodes.values()) {
+                if (input.equals(gm.toString().toLowerCase()) || input.equals(gm.toString())) {
```

Comment:
Gamemodes should be GameModes.

what happens if _input_ isn't one of the Gamemodes?